src/shared/components/mutationTable/column/AnnotationColumnFormatter.tsx

```diff
@@ -6,30 +6,37 @@ import OncokbPubMedCache from "shared/cache/PubMedCache";
 import CancerHotspots from "shared/components/annotation/CancerHotspots";
 import MyCancerGenome from "shared/components/annotation/MyCancerGenome";
 import OncoKB from "shared/components/annotation/OncoKB";
+import Civic from "shared/components/annotation/Civic";
 import {IOncoKbData} from "shared/model/OncoKB";
 import {IMyCancerGenomeData, IMyCancerGenome} from "shared/model/MyCancerGenome";
 import {IHotspotData} from "shared/model/CancerHotspots";
 import {Mutation} from "shared/api/generated/CBioPortalAPI";
 import {IndicatorQueryResp, Query} from "shared/api/generated/OncoKbAPI";
 import {generateQueryVariantId, generateQueryVariant} from "shared/lib/OncoKbUtils";
 import {isHotspot, is3dHotspot} from "shared/lib/AnnotationUtils";
+import {ICivicVariant, ICivicGene, ICivicEntry, ICivicVariantData, ICivicGeneData} from "shared/model/Civic.ts";
 
 export interface IAnnotationColumnProps {
     enableOncoKb: boolean;
     enableMyCancerGenome: boolean;
     enableHotspot: boolean;
+    enableCivic: boolean;
     hotspots?: IHotspotData;
     myCancerGenomeData?: IMyCancerGenomeData;
     oncoKbData?: IOncoKbData;
     oncoKbEvidenceCache?: OncoKbEvidenceCache;
     pubMedCache?: OncokbPubMedCache;
+    civicGenes?: ICivicGene;
+    civicVariants?: ICivicVariant;
 }
 
 export interface IAnnotation {
     isHotspot: boolean;
     is3dHotspot: boolean;
     myCancerGenomeLinks: string[];
     oncoKbIndicator?: IndicatorQueryResp;
+    civicEntry?: ICivicEntry | null;
```

Comment:
I need to differentiate here between three "states" for `civicEntry`, and this is how I did it:
- `undefined`: the request has not been finished, so we still don't know if the gene will have a `civicEntry` or not.
- `ICivicEntry`: the request has been finished and we have Civic data for the gene to be displayed.
- `null`: the request has been finished, and there is no Civic data for the gene to be displayed.